src/stacked_borrows.rs

```diff
@@ -151,6 +151,12 @@ impl<'tcx> Stack {
     /// Returns the index of the item we matched, `None` if it was the frozen one.
     /// `kind` indicates which kind of reference is being dereferenced.
     fn deref(&self, bor: Borrow, kind: RefKind) -> Result<Option<usize>, String> {
+        // Exclude unique ref and frozen tag.
+        match (kind, bor) {
+            (RefKind::Unique, Borrow::Shr(Some(_))) =>
+                return Err(format!("Encountered mutable reference with frozen tag")),
```

Comment:
The previous error reported some info on the pointer. Might be beneficial here, too.